src/crimson/os/seastore/segment_manager.cc

```diff
@@ -74,7 +74,7 @@ LOG_PREFIX(SegmentManager::get_segment_manager);
 	    ceph_assert(f);
 	    return f.ioctl(BLKGETNRZONES, (void *)&nr_zones);
 	  });
-      }).then([&](auto ret) -> crimson::os::seastore::SegmentManagerRef {
+      }).then([=, &nr_zones](auto ret) -> crimson::os::seastore::SegmentManagerRef {
```

Comment:
should capture by value from line 66 the first capture place, right?

Reply:
Looks error-prone. Let's make it explicit about which values are copied and which are captured by reference, replacing the default captures `=` and `&`.

Reply:
Changed to use capture-by-reference at all places, this seems to be the correct and much simpler way of fixing this.

Reply:
I'm with @cyx1231st, let's capture the variable explicitly rather than using a capture default.

Reply:
Thanks for the feedback. I have now changed it to explicitly use the variables instead of default capture. Please take a look.